src/OrbitGl/App.cpp

```diff
@@ -1083,8 +1084,34 @@ void OrbitApp::StartCapture() {
 
   CHECK(capture_client_ != nullptr);
 
-  auto capture_event_processor =
-      CaptureEventProcessor::CreateForCaptureListener(this, std::move(frame_track_function_ids));
+  std::unique_ptr<CaptureEventProcessor> capture_event_processor;
+  auto capture_for_listener_processor =
+      CaptureEventProcessor::CreateForCaptureListener(this, frame_track_function_ids);
+
+  if (absl::GetFlag(FLAGS_enable_capture_autosave)) {
```

Comment:
This whole if else tree would be more readable if there are early returns. Please consider extracting it to an extra method so you can use return